doc/zk-integration.md

```diff
@@ -0,0 +1,214 @@
+integrate with Espresso. There are a number of alternatives for such integration, depending on specific constraints the
+rollup may have such as using a particular data availability layer in addition to Tiramisu, or the need to update their
+state at a faster pace than the Espresso light client contract does.
+
+In the following, we assume that the rollup relies on Tiramisu for data availability and explore two different
+integrations flavors. With integration 1, the rollup relies on the Espresso light client contract to fetch the latest
+Espresso state updates. while with integration 2 the rollup checks Espresso consensus directly inside its circuit.
+
+For both integrations:
+
+- We describe the high level structure of the circuit as well as the rollup L1 contract and how they are related.
+- We describe the use of an Espresso liveness escape hatch in the rollup L1 contract.
+
+When the escape hatch is activated, both integrations use the same circuit  
+ which corresponds to a mode where the rollup falls back to using its own (centralized) sequencer as shown in Figure 1.
+
+![image](zk-rollup-default-sequencer.svg)
+
+**Figure 1:** Circuit used when the rollup falls back to using its default (centralized) sequencer. The public inputs
+are the current state of the rollup _COMM_STATE_VM i_, the new rollup state after update _COMM_STATE_VM i+1_ and a
+commitment to the transactions applied to the state, _COMM_TXS_ROLLUP_. The private input (in bold) corresponds the list
+of transactions.
+
+## Integration 1: Rollup contract fetches Espresso block commitment from the Espresso light client contract
+
+For this integration, Espresso consensus verification is delegated to the Espresso light client contract. In practice
+the rollup contract will fetch the last Espresso block commitment and feed it to the circuit. Still additional gadgets
+need to be introduced in order to implement the derivation pipeline logic consisting at a high level in:
+
+- Collecting all the Espresso commitments since the last update.
+- For each of these commitments, filter the corresponding Espresso blocks in order to obtain the transactions belonging
+  to the rollups.
+- Establishing some equivalence between the commitment to the rollup transactions used in the zkVM and the commitment to
+  those same transactions obtained after namespace filtering.
+
+![image](zk-rollup-circuit-no-espresso-consensus.svg) **Figure 2:** Rollup circuit with additional gadgets _Collect &
+Filter_, and _COMMs Equivalence_. Private inputs of the circuit are written in bold font.
+
+The circuit depicted in Figure 2 operates as follows:
+
+- The _Collect & Filter_ gadget receives as input _BLOCK_COMM_ESP_NEW_ which is the commitment to the latest Espresso
+  block available and _BLOCK_COMM_ESP_OLD_. Both of these commitments are public inputs. The first witness of this
+  circuit is _COMM_TXS_HISTORY_ESP_ which is a commitment to all the rollup transactions that have been sequenced since
```

Comment:
The rollup needs to compile this list and compute this commitment, correct?

Reply:
That is correct. Do you think it should be mentioned explicitly?

Reply:
I think so. Maybe just add a parenthetical sentence? Something like:

> The first witness of this circuit is _COMM_TXS_HISTORY_ESP_ which is a commitment to all the rollup transactions that have been sequenced since the last Espresso state update _BLOCK_COMM_ESP_OLD_. (_COMM_TXS_HISTORY_ESP_ is a rollup-specific commitment, so each rollup's prover must compute this commitment [as part of its derivation pipeline?] in order to proceed.)

Reply:
Done in ea195409.